front50-api/src/main/java/com/netflix/spinnaker/front50/api/model/pipeline/Trigger.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.front50.api.model.pipeline;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.lang.Object;
+import java.util.Set;
+
+
+abstract class ForwardingMap<K, V> implements Map<K, V> {
+  public ForwardingMap() {}
+
+  protected abstract Map<K, V> delegate();
+
+  @Override
+  public int size() {
+    return delegate().size();
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return delegate().isEmpty();
+  }
+
+  @Override
+  public boolean containsKey(Object key) {
+    return delegate().containsKey(key);
+  }
+
+  @Override
+  public boolean containsValue(@Nullable Object value) {
+    return delegate().containsValue(value);
+  }
+
+  @Override
+  public V get(Object key) {
+    return delegate().get(key);
+  }
+
+  @Override
+  public V put(K key, V value) {
+    return delegate().put(key, value);
+  }
+
+  @Override
+  public V remove(Object key) {
+    return delegate().remove(key);
+  }
+
+  @Override
+  public void putAll(Map<? extends K, ? extends V> m) {
+    delegate().putAll(m);
+  }
+
+  @Override
+  public void clear() {
+    delegate().clear();
+  }
+
+  @Override
+  public Set<K> keySet() {
+    return delegate().keySet();
+  }
+
+  @Override
+  public Collection<V> values() {
+    return delegate().values();
+  }
+
+  @Override
+  public Set<Entry<K, V>> entrySet() {
+    return delegate().entrySet();
+  }
+
+  @Override
+  public boolean equals(@Nullable Object object) {
+    return object == this || delegate().equals(object);
+  }
+
+  @Override
+  public int hashCode() {
+    return delegate().hashCode();
+  }
+
+}
+
+public class Trigger extends ForwardingMap<String, Object> {
```

Comment:
Needed to implement our own `ForwardingMap` in order to remove `com.google.common.collect.ForwardingMap;`. Removed the `COLLECTION_TYPE` since that was unnecessary since the `TriggerDeserializer` knows what type it is.